### PR TITLE
[v0.22.0] Backport PR 563

### PR DESCRIPTION
**Pull request by cardil**
Backporting knative-sandbox#563. And knative-sandbox#476 as it was required, too.

Original backport was done to 0.21 release in #215

It uses afd67ca to restore manually changed vendor files.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -38,3 +38,5 @@ require (
 	knative.dev/networking v0.0.0-20210331064822-999a7708876c
 	knative.dev/pkg v0.0.0-20210331065221-952fdd90dbb0
 )
+
+replace knative.dev/eventing => github.com/cardil/eventing v0.11.1-0.20210609102549-481a26fd9454
```
Comment (cardil) [inline go.mod:42]: /hold

This PR requires openshift/knative-eventing#1309 to be merged first. Then this should be replaced.

Comment (cardil) [reply go.mod]: This is now removed as openshift/knative-eventing#1309 is merged

/unhold

File: test/e2e/channel_subscription_ready_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"knative.dev/eventing-kafka/test/e2e/helpers"
+)
+
+func TestChannelSubscriptionScaleReadyV1(t *testing.T) {
+	helpers.ChannelSubscriptionScaleReadyHelper(context.Background(), t, channelTestRunner)
```
Comment (matzew) [inline test/e2e/channel_subscription_ready_test.go:30]: upstream is like:
https://github.com/knative-sandbox/eventing-kafka/blob/main/test/e2e/channel_subscription_ready_test.go#L30

Comment (cardil) [reply test/e2e/channel_subscription_ready_test.go:30]: That skip is a later thing. Done by @devguyio in https://github.com/knative-sandbox/eventing-kafka/pull/573

Here, I'm only back porting knative-sandbox#476 as it required by knative-sandbox#563

You think we should backport https://github.com/knative-sandbox/eventing-kafka/pull/573 as well?

Comment (cardil) [reply test/e2e/channel_subscription_ready_test.go:30]: There is also yours https://github.com/knative-sandbox/eventing-kafka/pull/685 @matzew, but you have abandoned it apparently :disappointed:

Comment (matzew) [reply test/e2e/channel_subscription_ready_test.go:30]: I closed the PR, b/c not yet possible to remove the skip: https://issues.redhat.com/browse/SRVKE-772

Comment (devguyio) [reply test/e2e/channel_subscription_ready_test.go:30]: we need to still skip the test, so yeah we need to backport knative-sandbox/eventing-kafka#573 .

Comment (cardil) [reply test/e2e/channel_subscription_ready_test.go:31]: I've added the knative-sandbox#573 commit as well.

**Comment by cardil** [comment]
/retest

Infra

**Pull request merged**